tests/test_data_management.py

```diff
@@ -155,3 +155,53 @@ def test_load_results(self, mock_file: MagicMock, mock_makedirs: MagicMock):
 
         mock_file.assert_called_once_with(path, "r")
         mock_makedirs.assert_not_called()
+
+
+class TestPlatformSerializationWithRuamelYaml:
+    """Unit tests to check the serialization is keeping all we want to keep."""
+
+    original_platform = ql.build_platform(Galadriel.runcard)
+    saved_platform = ql.build_platform(save_platform(path="./test.yml", platform=original_platform))
```

Comment:
Tests shouldn't create files! We should remove this.

Reply:
There should be a way of channeling `yaml.dump` into a variable instead of a file

Reply:
What about removing the file in the end, as I just did, with `os.remove(path)`?

I've seen this from some other parts in qililab, is that acceptable?

Reply:
I prefer not do it. I think you already had a bad experience with tests creating files 😆

Reply:
You can add a stream like you did in other cases!!

Reply:
Hahha you convinces me already with the first statement 🤣